privacyidea/lib/task/eventcounter.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#  2018-08-01 Cornelius Kölbel <cornelius.koelbel@netknights.it>
+#             EventCounter Task
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+import logging
+from privacyidea.lib.task.base import BaseTask
+from privacyidea.lib.monitoringstats import write_stats
+from privacyidea.lib.counter import reset, read
+from privacyidea.lib.utils import is_true
+
+
+__doc__ = """This task module reads event counters and writes them to the 
+MonitoringStats database table."""
+
+log = logging.getLogger(__name__)
+
+
+class EventCounterTask(BaseTask):
```

Comment:
In order to use this task class with the PeriodicTask API and cron runner, we just need to add the class to the TASK_MODULES list in lib/periodictask.py :-)
https://github.com/privacyidea/privacyidea/blob/d612000bdea07247917ef72a172afd5a1e3f7039/privacyidea/lib/periodictask.py#L35

Reply:
Will do so...